app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -268,6 +268,10 @@ def show_vm
         end
         attributes = @vm.attributes.deep_symbolize_keys
         attributes[:provider] = @compute_resource.provider
+        if attributes[:provider] == "Vmware"
+          attributes[:mac_addresses] = @vm.mac_addresses
```

Comment:
I'm using `attributes[:mac] = @vm.mac`, for displaying  the mac address in [hammer output](https://github.com/theforeman/hammer-cli-foreman/pull/624/commits/4c61a95ba691f5a854b11912a3e43a24801af0c9#diff-74c86b16ddb8d37dd475f05503a226bf55bf375e438c03a17c96bb2f22d9b29eR83), the same method is used for UI as well.
 
 With `attributes[:mac_addresses] = @vm.mac_addresses `  , I'm overriding the "mac_addresses" param of attributes for the API response which otherwise would be '{}'.  I explained why [here](https://matrix.to/#/!dxfWqeLNsImROheGXo:matrix.org/$170263335225036PRONT:matrix.org?via=matrix.org&via=ctrl-c.liu.se&via=sibnsk.net). 
 
 also mac returns the value in the string format `eg: "00:50:56:88:11:9e"`
 and mac_addresses returns it as a hash  `eg:{"Network Adapter 1"=> "00:50:56:88:11:9e"}`
 I suppose I could just fetch only one value (mac_addresses) but that would need me to format it to a string somewhere in the hammer_cli_foreman plugin.

Reply:
after the discussion, I looked at other compute resources and there's no good abstraction. Libvirt reports macs for all nics under key "nics" in a hash.

I start leaning towards "get what's available" approach. 

FTR, the mac is not among attributes, since it's implemented as a method https://github.com/fog/fog-vsphere/blob/ddc5f494e72b15de375900d0c87afd9fdcde8b9c/lib/fog/vsphere/models/compute/server.rb#L218-L220

I think this is good to go and the additional mac_addresses field can be only useful (today it was empty)